### PR TITLE
Add login form component

### PR DESCRIPTION
**Pull request by cnasc**
# Description

A piece of #2721, this adds the login form component.

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/LogIn.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
+// eslint-disable-next-line no-unused-vars
+import React, { FormEvent } from 'react'
+import styled from 'styled-components'
+import { connect } from 'react-redux'
+// eslint-disable-next-line no-unused-vars
```
Comment (cellog) [inline unlock-app/src/components/interface/LogIn.tsx]: this should not be necessary - the fix is to copy the config inside paywall to the top-level eslintrc:

```js
  rules: {
    'no-unused-vars': 'off',
    '@typescript-eslint/no-unused-vars': [
      'error',
      {
        vars: 'all',
        args: 'after-used',
        ignoreRestSiblings: true,
      },
    ],
```

Obviously, I am suggesting a quick follow-up PR, not changing this one

File: unlock-app/src/components/interface/LogIn.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+interface State {
+  emailAddress: string
+  password: string
+  submitted: boolean
+}
+
+export class LogIn extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    /* eslint-disable react/no-unused-state */
```
Comment (cellog) [inline unlock-app/src/components/interface/LogIn.tsx]: not sure but this also may be fixed?

File: unlock-app/src/components/interface/LogIn.tsx (hunk 1 above)
Comment (cnasc) [reply unlock-app/src/components/interface/LogIn.tsx]: Do these get turned off for everything? The issue is that eslint can't seem to correctly handle these in TS files, but we do want the coverage everywhere else.

Comment (cellog) [reply unlock-app/src/components/interface/LogIn.tsx]: `@typescript-eslint/no-unused-vars` catches unused variables, so we get the best of both worlds

Comment (cnasc) [reply unlock-app/src/components/interface/LogIn.tsx]: I mean, the typescript compiler also catches unused variables thanks to our TSConfig.

**Pull request merged**